server/polar/account/service.py

```diff
@@ -14,36 +14,67 @@
 from polar.postgres import AsyncSession
 from polar.enums import AccountType
 from polar.integrations.stripe.service import stripe
+from polar.integrations.open_collective.service import (
+    open_collective,
+    OpenCollectiveAPIError,
+    CollectiveNotFoundError,
+)
 
 
 log = structlog.get_logger()
 
 
+class AccountServiceError(Exception):
+    def __init__(self, message: str) -> None:
+        self.message = message
+        super().__init__(message)
+
+
+class AccountAlreadyExistsError(AccountServiceError):
+    def __init__(self) -> None:
+        super().__init__("An account already exists for this organization.")
+
+
 class AccountService(ResourceService[Account, AccountCreate, AccountUpdate]):
     async def create_account(
         self,
         session: AsyncSession,
         organization_id: UUID,
         admin_id: UUID,
         account: AccountCreate,
-    ) -> Account | None:
+    ) -> Account:
         existing = await self.get_by(session=session, organization_id=organization_id)
         if existing is not None:
-            return None
+            raise AccountAlreadyExistsError()
```

Comment:
I know I've said it 2 times already, but it's beautiful to see the enhanced error handling and especially how it leverages pydantic + early exceptions.